### PR TITLE
mise: 2023.12.35 -> 2024.1.11 rebrand from rtx

### LINKED ISSUE #278977: Package Request — mise
## Project Description

`mise` (pronounced "meez") or "mise-en-place" is a development environment setup tool. The name refers to a French culinary phrase that roughly translates to "setup" or "put in place". The idea is that before one begins cooking, they should have all their utensils and ingredients ready to go in their place.

> Notice there is already a package for this tool, under the former name: `rtx`. Some [history](https://mise.jdx.dev/rtx.html) for the name change.

## Metadata

* Homepage URL: https://mise.jdx.dev/rtx.html
* Source URL: https://github.com/jdx/mise
* License: MIT
* Platforms: Linux, Darwin

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by konradmalik**
## Description of changes

`rtx` was rebranded upstream to `mise`. See: https://mise.jdx.dev/rtx.html.

This PR renames the package, creates an alias and updates it to 2024.1.11

Needed to add one new patchShebangs path for `test_task_run` (`./test/cwd/.mise/tasks/filetask`). Without it, this test failed in checkPhase.

Closes #278977 

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/tools/misc/mise/default.nix [hunk 1]
```diff
@@ -51,25 +55,25 @@ rustPlatform.buildRustPackage rec {
...
-    tests.version = testers.testVersion { package = rtx; };
+    tests.version = testers.testVersion { package = mise; };
   };
 
   meta = {
-    homepage = "https://github.com/jdx/rtx";
-    description = "Polyglot runtime manager (asdf rust clone)";
-    changelog = "https://github.com/jdx/rtx/releases/tag/v${version}";
+    homepage = "https://mise.jdx.dev";
+    description = "dev tools, env vars, task runner";
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/mise/default.nix:73]:
```suggestion
    description = "The front-end to your dev env";
```
That's probably a bit more descriptive

File: pkgs/top-level/aliases.nix [hunk 2]
```diff
@@ -895,6 +895,7 @@ mapAliases ({
   rr-unstable = rr; # Added 2022-09-17
   rtl8723bs-firmware = throw "rtl8723bs-firmware was added in mainline kernel version 4.12"; # Added 2023-07-03
   rtsp-simple-server = throw "rtsp-simple-server is rebranded as mediamtx, including default config path update"; # Added 2023-04-11
+  rtx = throw "rtx was rebranded to mise"; # Added 2024-01-05
```
Comment (SuperSandro2000) [inline pkgs/top-level/aliases.nix:898]: Is this drop in? see https://github.com/jdx/mise#migrating-from-rtx

```suggestion
  rtx = mise; # Added 2024-01-05
```

**Review by SuperSandro2000 — APPROVED** [review]

File: pkgs/tools/misc/mise/default.nix (hunk 1 above)
Comment (konradmalik) [reply pkgs/tools/misc/mise/default.nix]: updated

File: pkgs/top-level/aliases.nix (hunk 2 above)
Comment (konradmalik) [reply pkgs/top-level/aliases.nix]: updated

**Comment by onemoresuza** [comment]
# Reviewed points

Result of `nixpkgs-review pr 279158` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>mise</li>
  </ul>
</details>

- [X] package name fits guidelines
- [X] package version fits guidelines
- [X] package builds on x86_64-linux
- [X] executables tested on x86_64-linux
- [X] all depending packages build
- [ ] patches have a comment describing either the upstream URL or a reason why the patch wasn't upstreamed
	- N/A: No patches are applied
- [ ] patches that are remotely available are fetched rather than vendored
	- N/A: No patches are applied
- [X] Package rename ensures an addition of the old name to `pkgs/top-level/aliases.nix`

**Review by onemoresuza — APPROVED** [review]

**Pull request merged**